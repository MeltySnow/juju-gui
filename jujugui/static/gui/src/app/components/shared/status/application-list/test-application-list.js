/* Copyright (C) 2018 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');

const StatusApplicationList = require('./application-list');

describe('StatusApplicationList', () => {
  let applications;

  const renderComponent = (options = {}) => enzyme.shallow(
    <StatusApplicationList
      applications={options.applications || applications}
      changeState={options.changeState || sinon.stub()}
      generateApplicationClickState={
        options.generateApplicationClickState || sinon.stub().returns('http://example.com')}
      generateCharmURL={options.generateCharmURL || sinon.stub()}
      generatePath={options.generatePath || sinon.stub()}
      onCharmClick={options.onCharmClick || sinon.stub()}
      statusFilter={options.statusFilter} />
  );

  beforeEach(() => {
    applications = [{
      getAttrs: sinon.stub().returns({
        charm: '~who/xenial/django-42',
        exposed: true,
        icon: 'django.svg',
        id: 'django',
        name: 'django',
        pending: false,
        status: {current: 'active'},
        units: [],
        workloadVersion: '1.10'
      })
    }];
  });

  it('renders', () => {
    const wrapper = renderComponent();
    expect(wrapper).toMatchSnapshot();
  });
});
