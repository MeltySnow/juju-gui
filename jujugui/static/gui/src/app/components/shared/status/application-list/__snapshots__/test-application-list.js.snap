// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusApplicationList renders 1`] = `
<StatusTable
  changeState={[Function]}
  generatePath={[Function]}
  headers={
    Array [
      Object {
        "columnSize": 2,
        "content": "Application",
      },
      Object {
        "columnSize": 2,
        "content": "Version",
      },
      Object {
        "columnSize": 2,
        "content": "Status",
      },
      Object {
        "columnSize": 1,
        "content": "Scale",
      },
      Object {
        "columnSize": 2,
        "content": "Charm",
      },
      Object {
        "columnSize": 2,
        "content": "Store",
      },
      Object {
        "columnSize": 1,
        "content": "Rev",
      },
    ]
  }
  rows={
    Array [
      Object {
        "classes": Array [
          "status-table__row--ok",
        ],
        "clickState": "http://example.com",
        "columns": Array [
          Object {
            "columnSize": 2,
            "content": <span>
              <img
                className="status-view__icon"
                src="django.svg"
              />
              django
            </span>,
          },
          Object {
            "columnSize": 2,
            "content": "1.10",
          },
          Object {
            "columnSize": 2,
            "content": <StatusLabel
              status="active"
            />,
          },
          Object {
            "columnSize": 1,
            "content": 0,
          },
          Object {
            "columnSize": 2,
            "content": <a
              className="status-view__link"
              onClick={[Function]}
            >
              u/who/django/xenial
            </a>,
          },
          Object {
            "columnSize": 2,
            "content": "jujucharms",
          },
          Object {
            "columnSize": 1,
            "content": 42,
          },
        ],
        "extraData": "ok",
        "key": "django",
      },
    ]
  }
/>
`;
